mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -334,8 +330,75 @@ class GuidelineAdherence(BuiltInScorer):
             },
         ]
         mlflow.genai.evaluate(data=data, scorers=[english, clarify])
+    """
+
+    name: str = "guidelines"
+    guidelines: Union[str, list[str]]
+    required_columns: set[str] = {"inputs", "outputs"}
+
+    def __call__(
+        self,
+        *,
+        inputs: dict[str, Any],
+        outputs: Any,
+    ) -> Assessment:
+        """
+        Evaluate adherence to specified guidelines.
+
+        Args:
+            inputs: A dictionary of input data, e.g. {"question": "What is the capital of France?"}.
+            outputs: The response from the model, e.g. "The capital of France is Paris."
+
+        Returns:
+            An :py:class:`mlflow.entities.assessment.Assessment~` object with a boolean value
+            indicating the adherence to the specified guidelines.
+        """
+        return judges.meets_guidelines(
+            guidelines=self.guidelines,
+            context={
+                "request": parse_inputs_to_str(inputs),
+                "response": parse_output_to_str(outputs),
+            },
+            name=self.name,
+        )
+
+    def with_config(
+        self,
+        *,
+        name: str = "guidelines",
+        guidelines: Optional[list[str]] = None,
+    ) -> "Guidelines":
+        """
+        Get a new scorer instance with the given name and global guidelines.
+
+        Args:
+            name: The name of the scorer. Default is "guidelines".
+            guidelines: A list of global guidelines to be used for evaluation.
+                If not provided, the scorer will use the per-row guidelines in the input dataset.
+
+        Returns:
+            The updated Guidelines scorer instance.
 
-    **2. Per-Example Guidelines**
+        Example:
+
+        .. code-block:: python
+
+            from mlflow.genai.scorers import guidelines
+
+            is_english = guidelines.with_config(
+                name="is_english", guidelines=["The response must be in English"]
+            )
+
+            mlflow.genai.evaluate(data=data, scorers=[is_english])
+        """
+        return Guidelines(name=name, guidelines=guidelines)
+
+
+@experimental
+class ExpectationsGuidelines(BuiltInScorer):
+    """
+    Guideline adherence evaluates whether the agent's response follows specific constraints
+    or instructions provided in the guidelines.
```

Comment:
```suggestion
    This scorer evaluates whether the agent's response follows specific constraints
    or instructions provided for each row in the evaluation dataset.
```

nit: also let's update L403 to sth like "To use this scorer, the input dataset should contain `expectations` column with `guidelines` field....`